### ISSUE TITLE
bug(monkey.bank): resonance bank double/triple-counts trade exits on container restart — kernel reward signal corrupted

### ISSUE THREAD

**Issue by GaryOcean428** [GaryOcean428/poloniex-trading-platform#574]
## Problem

`Monkey.bank.promoteToResonanceBank()` and the paired `reward pushed` writes fire **multiple times for the same orderId** during/after every container restart. The kernel is being trained on the same loss event 2-3× per deploy, biasing the loss tail of the resonance bank and the reward signal feeding the QLoRA training surface.

Silent, systematic, and present on every prior deploy.

## Evidence

Captured during the deploy of commit `558667a3` (Phase B sl_ratio change), 2026-04-27 ~15:05 UTC. The bug is **unrelated** to that commit — it surfaced because we were watching logs closely.

### Pass 1 — at exit time (15:05:21Z, before deploy lands)
Same orderId promoted twice, milliseconds apart, with both Position and Swing kernels pushing reward:

```
15:05:21.793Z  [Monkey.bank] promoted to resonance bank {symbol:ETH, pnl:-0.4527, depth:0.183}
15:05:21.793Z  [Monkey] witnessExit → bank {orderId:571832166808645632, pnl:-0.4527}
15:05:21.794Z  [Monkey.Position] reward pushed {symbol:ETH, pnlFrac:-4.53%}
15:05:21.801Z  [Monkey.bank] promoted to resonance bank {symbol:ETH, pnl:-0.4527, depth:0.183}  ← DUPLICATE
15:05:21.801Z  [Monkey] witnessExit → bank {orderId:571832166808645632, pnl:-0.4527}            ← DUPLICATE
15:05:21.801Z  [Monkey.Swing] reward pushed {symbol:ETH, pnlFrac:-4.53%}

15:05:21.819Z  [Monkey.bank] promoted to resonance bank {symbol:BTC, pnl:-1.66491, depth:0.148}
15:05:21.819Z  [Monkey] witnessExit → bank {orderId:571831537184890880, pnl:-1.6649}
15:05:21.819Z  [Monkey.Position] reward pushed {symbol:BTC, pnlFrac:-16.65%}
15:05:21.824Z  [Monkey.bank] promoted to resonance bank {symbol:BTC, pnl:-1.66491, depth:0.148}  ← DUPLICATE
15:05:21.824Z  [Monkey] witnessExit → bank {orderId:571831537184890880, pnl:-1.6649}            ← DUPLICATE
15:05:21.824Z  [Monkey.Swing] reward pushed {symbol:BTC, pnlFrac:-16.65%}
```

### Pass 2 — post-restart replay (15:05:53Z, container started 15:05:29)
After the new container started, both kernels replayed historical exits — including the two from Pass 1, AND six older orderIds that had presumably already been promoted in their original container's lifetime:

```
15:05:53.250Z  promoted ETH 571814891225432064 pnl:0.0804  (entry 13:48:13Z — hours earlier)
15:05:53.274Z  promoted ETH 571814891225432064 pnl:0.0804  ← DUPLICATE
15:05:53.275Z  promoted BTC 571743036296740864 pnl:0      (entry 09:02:42Z — 6h earlier)
15:05:53.276Z  promoted ETH 571832166808645632 pnl:-0.4527 ← THIRD WRITE for same orderId
15:05:53.277Z  promoted BTC 571831537184890880 pnl:-1.6649 ← THIRD WRITE for same orderId
... (6 more historical orderIds promoted, each twice for Position+Swing)
```

Eight unique orderIds got promoted again. Two of them (the recent losses) had already been written twice at exit time, so they hit the bank **three times each**.

### Pattern repeats on every restart
The deploy window 15:05–15:08 had multiple container stop/starts (15:05:28, 15:05:47, 15:06:00, 15:07:45, 15:07:54). The bank-replay block at 15:05:53 fires once per startup of the trading engine. Each restart re-replays history.

## Impact

1. **Loss tail of resonance bank is over-weighted by 2-3× per deploy.** Loss events get 2-3 entries; in-session wins get 1.
2. **Reward signal (`Monkey.Position.reward_pushed` / `Monkey.Swing.reward_pushed`) fires duplicate negative dop/ser/endo updates on every duplicate.**
3. **QLoRA training data is silently biased toward losses** — every `reward_pushed` event is what trains the kernel.
4. **Bank depth metric (`depth: 0.148`) is unreliable** because the same orderId can occupy multiple slots.

This has been happening on every deploy since witnessExit-replay-on-startup was added.

## Suspected root cause

Two suspects, possibly both:

1. **Both kernels (`monkey-position` and `monkey-swing`) write to the same shared bank** without coordinating. Each `promoteToResonanceBank()` call gets fired by both, with no orderId-level dedup.
2. **`witnessExit → bank` on startup replays exchange history** without checking whether each orderId has already been promoted in a prior session. Since the bank survives restart (it's persisted), but the replay logic doesn't query existing bank contents before promoting.

## Proposed fix

In `apps/api/src/services/monkey/resonance_bank.ts` (or wherever `promoteToResonanceBank` lives), add an orderId-level dedup guard:

```ts
async promoteToResonanceBank(entry: ExitEntry): Promise<boolean> {
  if (await this.hasOrderId(entry.orderId)) {
    log.debug('[Monkey.bank] skip duplicate promotion', { orderId: entry.orderId });
    return false;
  }
  // ... existing promotion logic
}
```

And the same guard around `reward_pushed` so duplicate `Position` / `Swing` calls for the same orderId are coalesced into one reward event.

## Acceptance criteria

- [ ] Unit test: calling `promoteToResonanceBank` twice with the same `orderId` results in one bank entry, not two.
- [ ] Unit test: when both Position and Swing kernels see the same exit, only one reward signal is emitted.
- [ ] Integration test: simulating a container restart with bank populated, the startup `witnessExit` replay does not re-promote already-banked orderIds.
- [ ] Verified in production logs after deploy: no two `[Monkey.bank] promoted` entries within 100ms of each other for the same orderId.

## Out of scope

- Bank dedup at the *content* level (basin coords, pnl) — orderId is the unique key.
- Whether the bank should clear on regime transition / DREAM cycle — separate question, separate issue.
- Whether the historical-replay-on-startup is even desirable — it might be, but with dedup.

## Severity

Medium-high. No data loss, no exchange-side risk, no trade-execution corruption. But this directly degrades the kernel's learning signal — which is the entire point of the resonance bank existing. Every deploy makes the bank's loss/win ratio progressively more pessimistic.

## Cross-refs

- Surfaced during Phase B deploy of commit `558667a3` (sl_ratio + symbol-aware TP)
- Independent of #571 (qig-warp finding) and #572 (Phase C registry)
- Pairs naturally with adding a one-line `MODE_PROFILES` config dump on engine start (separate small follow-up — Issue TBD)

🤖 Issue authored by Claude Opus 4.7 from log evidence.
